caddyconfig/httpcaddyfile/httptype.go

```diff
@@ -376,45 +376,65 @@ func (st *ServerType) serversFromPairings(
 				return nil, fmt.Errorf("server block %v: compiling matcher sets: %v", sblock.block.Keys, err)
 			}
 
-			hosts := sblock.hostsFromKeys(false, false)
+			var httpHosts, httpsHosts []string
+			for _, key := range sblock.block.Keys {
+				addr, err := ParseAddress(key)
+				if err != nil {
+					return nil, err
+				}
+				addr = addr.Normalize()
 
-			// tls: connection policies
-			if cpVals, ok := sblock.pile["tls.connection_policy"]; ok {
-				// tls connection policies
-				for _, cpVal := range cpVals {
-					cp := cpVal.Value.(*caddytls.ConnectionPolicy)
-
-					// make sure the policy covers all hostnames from the block
-					for _, h := range hosts {
-						if h == defaultSNI {
-							hosts = append(hosts, "")
-							cp.DefaultSNI = defaultSNI
-							break
-						}
-					}
+				if addr.Scheme == "http" {
```

Comment:
No consideration for port? Semantically, http:// and port 80 are equivalent, as are https:// and :443.

Reply:
I don't think we need to be concerned with the port.So I tested with server hostnames like `:80` `http://:80`